spec/unit/puppet_x/sql_connection_spec.rb

```diff
@@ -20,11 +25,11 @@ def stub_connection
         @connection.stubs(:Open).with('Provider=SQLNCLI11;Initial Catalog=master;Application Name=Puppet;Data Source=.;DataTypeComptibility=80;User ID=sa;Password=Pupp3t1@')
       end
       it 'should not raise an error but populate has_errors with message' do
-        subject.stubs(:execute).raises(Exception.new("SQL Server\n error has happened"))
+        @connection.Errors.stubs(:count).returns(1)
```

Comment:
This seems an odd stub to add

Reply:
I have to have it there, or it never attempts to examine the @connection object for errors.

Reply:
This stub is fine... he has to simulate a populated ADODB Errors connection object, as the existence of errors in this collection changes the behavior of the error reporting.

It may be possible to create an instance of the real thing, but that would require Windows...

Reply:
The need to create and test the real thing is why I modified the acceptance tests to ensure the proper object is examined and just the win32 exception message passed into the rescue.

Reply:
I guess it "feels odd" to me that your stubbing the length of an array, but not setting the content of the Array too.

Reply:
Well, it's a strange object. Calling ```count``` on the ```Errors``` object isn't just asking for the length of the array as ``Errors``` isn't a plain array object. ```count``` is a real method on the object that has other properties and methods. So in this case I'm stubbing a method call. The content that gets returned is set higher up in the ```stub_connection``` helper at the top of the file. It defaults to zero, but in the case of tests where I actually want to take a look at the error text in the object I have to override that and return 1 as the count.

Reply:
@glennsarti yeah, higher up he sets `error_mock.stubs(:Description).returns("SQL Error in Connection")` ... but since the default is 0 items, I agree it should probably be done where the length is stubbed to 1 to make things clearer.

Reply:
The two test cases where this was used have been modified to set the non default ```count``` value and the ```Description``` value inside the test case.